src/coord/src/session.rs

```diff
@@ -136,10 +136,13 @@ impl Session {
             | TransactionStatus::InTransaction(txn_ops)
             | TransactionStatus::InTransactionImplicit(txn_ops) => match txn_ops {
                 TransactionOps::None => *txn_ops = add_ops,
-                TransactionOps::Reads => match add_ops {
-                    TransactionOps::Reads => {}
+                TransactionOps::Peeks(txn_ts) => match add_ops {
+                    TransactionOps::Peeks(add_ts) => {
+                        assert_eq!(*txn_ts, add_ts);
```

Comment:
this assert makes me kinda nervous -- I know there's only two callsites for `add_transaction_ops` and we check to give an equivalent timestamp in the peek case. not sure how much work it would be to convert this to an error

Reply:
I hear you, but I think the assert IS supposed to make us nervous. Any error returned here would have to be of the form "oops! we messed up, this is an mz bug, not yours" because mz should never produce this situation. The error paths in this function are all for user-visible things. If it asserts, I feel like that's ok because the user will hopefully report it to us and we can fix a real bug.